### PR TITLE
feat: job and app group filters revamp

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ApplicationGroup/AppGroup.service.ts [hunk 1]
```diff
@@ -189,27 +191,17 @@ export const getConfigAppList = (envId: number, appIds: string): Promise<ConfigA
...
-        const urlParams = Object.entries(params).map(([key, value]) => {
-            if (!value) {
-                return
-            }
-            return `${key}=${value}`
-        })
-        return get(`${Routes.ENVIRONMENT_APPS}?${urlParams.filter((s) => s).join('&')}`, options)
-    }
-    return get(Routes.ENVIRONMENT_APPS, options)
+    const url = getUrlWithSearchParams(Routes.ENVIRONMENT_APPS, {
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/AppGroup.service.ts:198]: Can we add type for params?

File: src/components/ApplicationGroup/AppGroup.service.ts [hunk 2]
```diff
@@ -189,27 +191,17 @@ export const getConfigAppList = (envId: number, appIds: string): Promise<ConfigA
...
-        envName?: string
-        clusterIds?: string
-        offset?: string
-        size?: string
-    },
-    signal?: AbortSignal,
-): Promise<EnvAppType> => {
+export const getEnvAppList = (filterConfig?: Partial<AppGroupFilterConfig>, signal?: AbortSignal): Promise<EnvAppType> => {
     const options = signal ? { signal } : null
+    const {searchKey, cluster, offset, pageSize} = filterConfig
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/AppGroup.service.ts:196]: If filterConfig is optional in params, we may need to add fallback before de-structuring

Comment (AbhishekA1509) [inline src/components/ApplicationGroup/AppGroup.service.ts:196]: Please format

File: src/components/Jobs/Service.ts [hunk 3]
```diff
@@ -35,64 +34,41 @@ export const getJobCIPipelines = (jobId: number) => {
     return get(`${Routes.JOB_CI_PIPELINE_LIST}/${jobId}`)
 }
 
-export const getJobsInitData = (payloadParsedFromUrl: Record<string, any>): Promise<any> => {
+export const getJobsInitFilters = (): Promise<JobsMasterFilters> => {
     return Promise.all([getProjectList(), getEnvironmentListMinPublic()]).then(([projectsRes, environmentsRes]) => {
```
Comment (AbhishekA1509) [inline src/components/Jobs/Service.ts:38]: Please use getEnvironmentListMinPublic from common, and please connect with @eshankvaish, if newly created resource filter hook can be used?

File: src/components/Jobs/Service.ts [hunk 4]
```diff
@@ -35,64 +34,41 @@ export const getJobCIPipelines = (jobId: number) => {
...
-                isChecked: filterApplied.appStatus.has(values),
-            }
-        })
-
-        filters.environments = (
-            environmentsRes.result
+            ).sort((a, b) => {
+                return sortOptionsByLabel(a, b)
+            }),
+            status: [
```
Comment (AbhishekA1509) [inline src/components/Jobs/Service.ts:51]: Can we move this to constant? and while using please use structuredClone here

File: src/components/Jobs/JobList/JobsList.tsx [hunk 5]
```diff
@@ -101,183 +146,106 @@ export default function JobsList() {
...
+                initialSearchText={searchKey}
+                containerClassName="dc__mxw-250 flex-grow-1"
+                handleEnter={handleSearch}
+                inputProps={{
+                    placeholder: 'Search by job name',
+                    autoFocus: true,
+                }}
+                dataTestId="Search-by-job-name"
+            />
+            <div className="flexbox dc__gap-8 dc__align-items-center dc__zi-20">
```
Comment (AbhishekA1509) [inline src/components/Jobs/JobList/JobsList.tsx:199]: Check if z-index can be minimised?

File: src/components/ApplicationGroup/AppGroup.service.ts [hunk 6]
```diff
@@ -189,27 +191,17 @@ export const getConfigAppList = (envId: number, appIds: string): Promise<ConfigA
...
-                return
-            }
-            return `${key}=${value}`
-        })
-        return get(`${Routes.ENVIRONMENT_APPS}?${urlParams.filter((s) => s).join('&')}`, options)
-    }
-    return get(Routes.ENVIRONMENT_APPS, options)
+    const url = getUrlWithSearchParams(Routes.ENVIRONMENT_APPS, {
+        envName: searchKey,
+        clusterIds: cluster?.join(),
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/AppGroup.service.ts:200]: Suggestion: There is support for array inside getUrlWithSearchParams please check if we can leverage that?

File: src/components/ApplicationGroup/List/EnvironmentsList.tsx [hunk 7]
```diff
@@ -14,125 +14,77 @@
...
-        })
-        if (!ids.length) {
-            queryParams.delete(type)
-        } else {
-            queryParams.set(type, ids.toString())
-        }
-        queryParams.set('offset', '0')
-        history.push(`${match.path}?${queryParams.toString()}`)
-    }
+    const [clusterListLoading, clusterListRes] = useAsync(() => getClusterListMinWithoutAuth())
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/List/EnvironmentsList.tsx:55]: can directly send getClusterListMinWithoutAuth

File: src/components/ApplicationGroup/List/EnvironmentsList.tsx [hunk 8]
```diff
@@ -14,125 +14,77 @@
...
-        const queryParams = new URLSearchParams(location.search)
-        queryParams.delete('cluster')
-        queryParams.delete('search')
-        queryParams.set('offset', '0')
-        setSearchText('')
-        history.push(`${match.path}?${queryParams.toString()}`)
+    const getFormattedValue = (filterKey: AppGroupUrlFilters , filterValue: string) => {
+        switch (filterKey) {
+            case AppGroupUrlFilters.cluster:
+                return clusterOptions.find((clusterItem) => clusterItem.value === filterValue)?.label
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/List/EnvironmentsList.tsx:71]: Are we sure we should that this, can we verify the case of deleted cluster name in url

File: src/components/ApplicationGroup/List/EnvironmentListView.tsx [hunk 9]
```diff
@@ -57,62 +57,38 @@ const EnvironmentLink = ({
...
-        envName: params.get('search'),
-        clusterIds: params.get('cluster'),
-        offset: params.get('offset') || '0',
-        size: params.get('pageSize') || '20',
-    }
-    const [paramsData, setParamsData] = useState(paramObj)
-    const [loading, appList] = useAsync(() => getEnvAppList(paramsData), [paramsData])
-    const emptyStateData = paramObj.clusterIds
+    const { cluster } = filterConfig
+    const [loading, appList] = useAsync(() => getEnvAppList(filterConfig), [filterConfig])
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/List/EnvironmentListView.tsx:70]: Please rename loading

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

File: src/components/ApplicationGroup/AppGroup.service.ts (hunk 6 above)
Comment (arunjaindev) [reply src/components/ApplicationGroup/AppGroup.service.ts:200]: we are sending cluster ids as comma seperated values here

File: src/components/ApplicationGroup/List/EnvironmentsList.tsx (hunk 8 above)
Comment (arunjaindev) [reply src/components/ApplicationGroup/List/EnvironmentsList.tsx:71]: working fine

File: src/components/ApplicationGroup/AppGroup.service.ts (hunk 2 above)
Comment (AbhishekA1509) [reply src/components/ApplicationGroup/AppGroup.service.ts]: Not yet resolved, we also need to filterConfig || {}

File: src/components/Jobs/JobList/JobsList.tsx [hunk 10]
```diff
@@ -101,183 +142,106 @@ export default function JobsList() {
...
-                <SearchBar
-                    initialSearchText={searchKey}
-                    containerClassName="dc__mxw-250 flex-grow-1"
-                    handleEnter={handleSearch}
-                    inputProps={{
-                        placeholder: 'Search by job name',
-                        autoFocus: true
-                    }}
-                    dataTestId="Search-by-job-name"
+    const renderMasterFilters = () => (
```
Comment (AbhishekA1509) [inline src/components/Jobs/JobList/JobsList.tsx:183]: We can move filters into a separate component

File: src/components/Jobs/Service.ts (hunk 3 above)
Comment (arunjaindev) [reply src/components/Jobs/Service.ts:38]: will discuss and pick later for resource hook

File: src/components/ApplicationGroup/List/EnvironmentListView.tsx [hunk 11]
```diff
@@ -57,62 +57,38 @@ const EnvironmentLink = ({
...
-    const history = useHistory()
+export default function EnvironmentsListView({
+    isSuperAdmin,
+    filterConfig,
+    clearFilters,
+    changePage,
+    changePageSize,
+}: EnvironmentsListViewType) {
     const [filteredEnvList, setFilteredEnvList] = useState<EnvAppList[]>([])
     const [envCount, setEnvCount] = useState<number>()
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/List/EnvironmentListView.tsx:68]: Can put default to 0

File: src/components/ApplicationGroup/List/EnvironmentListView.tsx [hunk 12]
```diff
@@ -57,62 +57,38 @@ const EnvironmentLink = ({
...
     useEffect(() => {
-        setParamsData(paramObj)
-    }, [location.search])
-
-    useEffect(() => {
-        if (appList?.result?.envList) {
-            setFilteredEnvList(appList.result.envList)
-            setEnvCount(appList.result.envCount)
-            setPaginationParamsChange({ pageSize: +params.get('pageSize') || 20, offset: +params.get('offset') })
+        if (appListResponse?.result?.envList) {
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/List/EnvironmentListView.tsx:76]: Suggestion:
There is no need of if else:
```
const appListResult = appListResponse?.result || {}
const envList = appListResult.envList ?? []
const envCount = appListResult.envCount ?? 0
```

File: src/components/ApplicationGroup/List/EnvironmentsList.tsx [hunk 13]
```diff
@@ -14,125 +14,75 @@
...
-        })
-        if (!ids.length) {
-            queryParams.delete(type)
-        } else {
-            queryParams.set(type, ids.toString())
-        }
-        queryParams.set('offset', '0')
-        history.push(`${match.path}?${queryParams.toString()}`)
-    }
+    const [clusterListLoading, clusterListRes] = useAsync(getClusterListMinWithoutAuth)
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/List/EnvironmentsList.tsx:55]: Do we need to handle error state?

**Review by AbhishekA1509 — CHANGES_REQUESTED** [review]

**Review by AbhishekA1509 — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**